### PR TITLE
fix: move cancelled upcoming booking to cancelled tab

### PR DESCRIPTION
**Pull request by mihaic195**

### CODE REVIEW

**Review by alishaz-polymath — APPROVED** [review]
Looks good to me

File: server/routers/viewer.tsx [hunk 1]
```diff
@@ -28,7 +28,7 @@ export const viewerRouter = createProtectedRouter()
       const { prisma, user } = ctx;
       const bookingListingByStatus = input.status;
       const bookingListingFilters: Record<typeof bookingListingByStatus, Prisma.BookingWhereInput[]> = {
-        upcoming: [{ endTime: { gte: new Date() } }],
+        upcoming: [{ endTime: { gte: new Date() }, NOT: { status: { equals: BookingStatus.CANCELLED } } }],
         past: [{ endTime: { lte: new Date() } }],
         cancelled: [{ status: { equals: BookingStatus.CANCELLED } }],
       };
```
Comment (KATT) [inline server/routers/viewer.tsx:34]: Can't we just use the `input.status` in the `where` of `bookingsQuery` below?

I.e.

```ts
      const bookingsQuery = await prisma.booking.findMany({
        where: {
          ...(input.status === 'upcoming' ? { /* specific filters for upcoming */ } : {}),
          ...(input.status === 'past' ? { /* specific filters for past */ } : {}),
          ...(input.status === 'cancelled' ? { /* specific filters for cancelled */ } : {}),
      },
    // ...
```

**Review by KATT — APPROVED** [review]
Ignore my first comment - deleted it - looks good.

File: server/routers/viewer.tsx [hunk 2]
```diff
@@ -28,7 +28,7 @@ export const viewerRouter = createProtectedRouter()
       const { prisma, user } = ctx;
       const bookingListingByStatus = input.status;
       const bookingListingFilters: Record<typeof bookingListingByStatus, Prisma.BookingWhereInput[]> = {
-        upcoming: [{ endTime: { gte: new Date() } }],
+        upcoming: [{ endTime: { gte: new Date() }, NOT: { status: { equals: BookingStatus.CANCELLED } } }],
         past: [{ endTime: { lte: new Date() } }],
```
Comment (KATT) [inline server/routers/viewer.tsx:32]: This need the same query right?

Comment (KATT) [inline server/routers/viewer.tsx:32]:
```suggestion
        past: [{ endTime: { lte: new Date() }, NOT: { status: { equals: BookingStatus.CANCELLED } } }],
```

**Review by KATT — CHANGES_REQUESTED** [review]
I think this is needed?

**Review by KATT — APPROVED** [review]

**Pull request merged**